src/app/api/api_v1/endpoints/detections.py

```diff
@@ -26,10 +26,14 @@
 async def create_detection(
     azimuth: float = Form(..., gt=0, lt=360, description="angle between north and direction in degrees"),
     file: UploadFile = File(..., alias="file"),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[Role.CAMERA]),
 ) -> Detection:
     telemetry_client.capture(f"camera|{token_payload.sub}", event="detections-create")
+    camera = await cameras.get(token_payload.sub, strict=True)
+    if camera is not None and token_payload.organization_id != camera.organization_id:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden.")
```

Comment:
Ah sorry, that might not have been clear: at this step, the token has already been verified as being a camera token. Because the camera is creating the detection, I don't think there is a need for the extra verification (i.e. the camera will stamp the detection with its org ID, or not even necessary since the detection has the cam ID)

In short, this verif can only failed if the token was forged (secret key hacked)